tesseract_motion_planners/ompl/src/ompl_motion_planner.cpp

```diff
@@ -60,43 +59,6 @@ using CachedSimpleSetupsPtr = std::shared_ptr<CachedSimpleSetups>;
 
 namespace tesseract_planning
 {
-bool checkStartState(const ompl::base::ProblemDefinitionPtr& prob_def,
-                     const Eigen::Ref<const Eigen::VectorXd>& state,
-                     const OMPLStateExtractor& extractor)
-{
-  if (!(prob_def->getStartStateCount() >= 1))
-    return false;
-
-  for (unsigned i = 0; i < prob_def->getStartStateCount(); ++i)
-    if (extractor(prob_def->getStartState(i)).isApprox(state, 1e-5))
-      return true;
-
-  return false;
-}
-
-bool checkGoalState(const ompl::base::ProblemDefinitionPtr& prob_def,
-                    const Eigen::Ref<const Eigen::VectorXd>& state,
-                    const OMPLStateExtractor& extractor)
-{
-  ompl::base::GoalPtr goal = prob_def->getGoal();
-  if (goal->getType() == ompl::base::GoalType::GOAL_STATE)
-    return extractor(prob_def->getGoal()->as<ompl::base::GoalState>()->getState()).isApprox(state, 1e-5);
-
-  if (goal->getType() == ompl::base::GoalType::GOAL_STATES)
-  {
-    auto* goal_states = prob_def->getGoal()->as<ompl::base::GoalStates>();
-    for (unsigned i = 0; i < goal_states->getStateCount(); ++i)
-      if (extractor(goal_states->getState(i)).isApprox(state, 1e-5))
-        return true;
-  }
-  else
-  {
-    CONSOLE_BRIDGE_logWarn("checkGoalStates: Unsupported Goal Type!");
-    return true;
-  }
-  return false;
-}
```

Comment:
Note, these functions were removed because they are RVSS-specific and don't really add any value at run-time IMO. If we're concerned that OMPL is not setting valid start and end states, we should either make a unit test to prove that this is not the case, or bake these checks in as real checks (not asserts) into the RVSS profile

Reply:
I think these should only be removed if a unit test is written in its place. I typically add assets for two reasons, the first if something is not easy to write a unit test and the second is I have not had time to write a unit test yet. They still have value in CI.